CMakeLists.txt

```diff
@@ -70,19 +70,32 @@ project(Daemon C CXX)
 ################################################################################
 
 if(NOT DAEMON_EXTERNAL_APP)
-    option(BUILD_CLIENT "Build Daemon client" 1)
-    option(BUILD_SERVER "Build Daemonded server" 1)
-    option(BUILD_GAME_NATIVE_EXE "Build native executable game logic" 1)
-    option(BUILD_GAME_NATIVE_DLL "Build native shared library game logic" 1)
+    option(BUILD_CLIENT "Engine client executable, required to play" ON)
+    option(BUILD_SERVER "Engine server executable, required to host servers" ON)
+
+    ## About the different ways to host/play games:
+    ## Native DLL: no sandboxing, no cleaning up but debugger support. Use for dev.
+    ## NaCl exe: sandboxing, no leaks, slightly slower, hard to debug. Use for regular players.
+    ## Native exe: no sandboxing, no leaks, hard to debug. Might be used by server owners for perf.
+    ## see VirtualMachine.h for code
+
+    # can be loaded by daemon with vm.[sc]game.type 0 or 1
+    # TODO (maybe): those could probably be merged as a single option "BUILD_DISTRIBUTED_FILES"
+    option(BUILD_GAME_NACL "Build the «pexe» files, required to build the «nexe» files which are required to host servers." ON)
+    option(BUILD_GAME_NACL_NEXE "Build the «nexe» files from the «pexe» files, required to host servers." ON)
```

Comment:
Why not make this one advanced? It is an extremely niche option.

Reply:
It seems less advanced than BUILD_DUMMY_APP to me, and also looks like something that should not be "hidden" into advanced (advanced options are really a mess, I'm not sure project's options should go there honestly)

Reply:
There is no reason for anyone to ever use it at this point, and it is confusing to have both optoins, so the user shouldn't have to see it. (At one point it might have been a basis for developing runtime pexe->nexe translation).  I have used BUILD_DUMMY_APP several times, but never this one.

Reply:
building the NEXE files is probably not really "advanced" as long as one wants to build a mod. Now, I have no idea what would happen with only one of those defined. I suppose there should be a single option enabling both of them?

Reply:
Building pexe without nexe is advanced. The user would have to modify BUILD_GAME_NACL_NEXE to do that. Only BUILD_GAME_NACL should be shown to the user since it is the one which has the expected effect of turning on/off both pexe and nexe (given that BUILD_GAME_NACL_EXE is on). So BUILD_GAME_NACL_NEXE is the one which should be marked as advanced.

Reply:
From "quick" testing:

* `BUILD_GAME_NACL_NEXE` requires `BUILD_GAME_NACL` to be of any use at all.
* `BUILD_GAME_NACL` generates only the `*.pexe`

So if I'm correct, what should be done is to only show `BUILD_GAME_NACL_NEXE` by default, and when it's enabled, enable `BUILD_GAME_NACL`. I would even say that being able to toggle NEXE without the other and without a warning  in cmake is a bug.

Reply:
There is a cmake_dependent_option function that does what you want (makes an option not show up unless another one is enabled). It has the serious drawback that the option is hidden (in ccmake/cmake-gui) until you hit "Configure" again. This can make an option hard to find, but since this option does not need to be seen by anyone, I guess it would be OK here...

Reply:
I gave it a try after reading cmake's doc and can't really understood how I got the results I had. This would have been fine if the thing what I wanted, but it does not, by far.

I suppose the better at this point I'll just do a big option which sets both the variables to the same value. Easy and reliable, at least. I can even change the name to something which speaks more this way.